Adding basic Swift support

AdamEisfeld opened the pull request:
- Adding SKTypeAliases.swift for bridging between closure and objective-c block
- Modifying README.md to contain Swift example / Swift usage instructions (also minor typo fixes)
- Modifying StateKit.podspec to use subspecs, allowing users not interested in Swift to not be required to update their Podfile (also incrementing version number)

Note: I didn't modify the Example's podspec to pull in the Swift subspec as there isn't any example code using it yet. Perhaps in the future there can be 2 examples, 1 in Swift and 1 in ObjC.

AdamEisfeld committed:
Adding basic Swift support:
- Adding SKTypeAliases.swift for bridging between closure and objective-c block
- Modifying README.md to contain Swift example / Swift usage instructions (also minor typo fixes)
- Modifying StateKit.podspec to use subspecs, allowing users not interested in Swift to not be required to update their Podfile (also incrementing version number)

AdamEisfeld commented:
Don't merge yet, I just noticed the typo in the Podspec

AdamEisfeld committed:
- Fixing typo in Podspec that had "ObjC" subspec using Swift name

AdamEisfeld committed:
- running pod update on Example

sghiassy commented:
Thx @AdamEisfeld for the PR. Is the PR ready to be reviewed? I remember seeing a comment in my email inbox saying to hold off until something was fixed first.

AdamEisfeld commented:
Yeah I commented to hold off on merging because I had accidentally declared the ObjC subspec with a typo. I fixed it and deleted the comment (sorry, should have just left the comment in and said I fixed it).

I also originally mentioned in the PR that I didn't run pod update on the example as I didn't think you'd want a bunch of commits from an update that don't really change the Example as I didn't add any Swift example usage, but I noticed the build failed because of that so I ended up running pod update on the Example's StateKit pod to fix.

So to sum up, yes it should be ready to be reviewed now. I had an idea to further change the README to include info on using [weak self] / [unowned self] in the chart blocks, as well as getting Swift support setup, but I was going to ask you today if that'd be better left until an actual Swift example was added, or just in a separate PR.

AdamEisfeld commented on Example/Pods/Target Support Files/Pods-StateKit/Pods-StateKit-acknowledgements.markdown:
```diff
@@ -3,24 +3,34 @@ This application makes use of the following third party libraries:
+PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
```
Hmmm... just noticed this got replaced with a 2013 Groupon copyright :/. Strange.

sghiassy commented:
Good looking PR. 

Checked:
- All tests: pass
- Example app runs: pass

Thanks for the effort!
